### PR TITLE
OPENTOK-43815: Check for invalid token role

### PR DESCRIPTION
**Pull request by v-kpheng**
#### What is this PR doing?
Checks for invalid token role.  If valid, a default is assumed.

#### How should this be manually tested?
Confirm that invalid token roles can cause meet to crash
* Checkout `master` branch
* Build and run meet
* Open http://localhost:3000/testing-if-meet-crashes?tokenRole=moderators
* Notice that meet crashes

Confirm that invalid token role no longer causes meet to crash
* Checkout this branch
* Build and run meet
* Open http://localhost:3000/testing-if-meet-crashes?tokenRole=moderators
* Notice that meet doesn't crash

#### What are the relevant tickets?
https://jira.vonage.com/browse/OPENTOK-43815

### CODE REVIEW

File: src/js/login/controller.js [hunk 1]
```diff
@@ -9,16 +10,24 @@ angular.module('opentok-meet-login', [])
     $scope.dtx = false;
     $scope.joinRoom = () => {
       let url = $window.location.href + encodeURIComponent($scope.room);
+
       if ($scope.roomType !== 'normal') {
         url += `/${$scope.roomType}`;
       }
+
+      if (!isValidTokenRole($scope.tokenRole)) {
+        $scope.tokenRole = 'moderator';
```
Comment (michaelwhittemore) [inline src/js/login/controller.js:19]: Why do we make the default tokeRole 'moderator' here, but make it 'publisher' in server/routes.js?

Comment (v-kpheng) [reply src/js/login/controller.js:19]: @michaelwhittemore, I just left it as-is.  I don't know why either :-(

Comment (v-kpheng) [reply src/js/login/controller.js:19]: @michaelwhittemore, I'll make it moderator throughout.

File: tests/unit/isValidTokenRoleSpec.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const tokenRoles = ['moderator', 'publisher', 'subscriber'];
+
+  tokenRoles.forEach((tokenRole) => {
+    it(`should return true since token role ${tokenRole} is valid`, () => {
+      expect(isValidTokenRole('moderator')).toBe(true);
+    });
+
+    const invalidTokenRole = `${tokenRole}s`;
+
+    it(`should return false since token role ${invalidTokenRole} is valid`, () => {
```
Comment (michaelwhittemore) [inline tests/unit/isValidTokenRoleSpec.js:13]: I think this should read `it(`should return false since token role ${invalidTokenRole} is *in*valid``

**Review by michaelwhittemore — APPROVED** [review]
LGTM

**Review by michaelwhittemore — APPROVED** [review]
Tested, invalid token roles no longer crash meet. LGTM

File: server/routes.js [hunk 3]
```diff
@@ -61,7 +65,7 @@ module.exports = (app, config, redis, ot, redirectSSL) => {
               apiKey: (pApiKey && pSecret) ? pApiKey : config.apiKey,
               p2p: RoomStore.isP2P(room),
               token: otSDK.generateToken(sessionId, {
-                role: tokenRole || 'publisher',
+                role: tokenRole || 'moderator',
```
Comment (maikthomas) [inline server/routes.js:68]: @v-kpheng 
This `|| 'moderator'`is never used, you have already put `'publisher'` as the default on line 44

File: tests/unit/login/controllerSpec.js [hunk 4]
```diff
@@ -84,6 +84,13 @@ describe('OpenTok Login Page', () => {
         scope.joinRoom();
         expect(windowMock.location.href).toEqual('mockURL/foo/screen?tokenRole=subscriber');
       });
+
+      it('defaults the url to moderator for screen roomType, when the tokenRole is invalid', () => {
```
Comment (maikthomas) [inline tests/unit/login/controllerSpec.js:88]: 👍

**Pull request merged**